picasso/src/main/java/com/squareup/picasso/MediaStoreRequestHandler.java

```diff
@@ -103,6 +107,37 @@ static PicassoKind getPicassoKind(int targetWidth, int targetHeight) {
   }
 
   static int getExifOrientation(ContentResolver contentResolver, Uri uri) {
+    int exifOrientation = getExitOrientationFromFile(contentResolver, uri);
+    if (exifOrientation == ExifInterface.ORIENTATION_UNDEFINED) {
+      exifOrientation = getExifOrientationFromContentResolver(contentResolver, uri);
+    }
+    return exifOrientation;
+  }
+
+  static int getExitOrientationFromFile(ContentResolver contentResolver, Uri uri) {
+    Cursor cursor = null;
+    try {
+      contentResolver.openInputStream(uri);
+      cursor = contentResolver.query(uri, CONTENT_DATA, null, null, null);
+      if (cursor == null || !cursor.moveToFirst()) {
+        return 0;
+      }
+      String filePath = cursor.getString(0);
+      ExifInterface exifInterface = new ExifInterface(filePath);
+      return exifInterface.getAttributeInt(ExifInterface.TAG_ORIENTATION,
+              ExifInterface.ORIENTATION_UNDEFINED);
+
+    } catch (Exception ignored) {
```

Comment:
Instead of catching and ignoring all exceptions, I think the proper way would be to retrieve the mime type of the image and only attempt to get orientation from exif metadata if the type supports exif (jpeg), otherwise return undefined.